VERSIONING.md

```diff
@@ -0,0 +1,220 @@
+# Versioning
+
+This document describes the versioning policy for this repository. This policy
+is designed so the following goals can be achieved.
+
+**Users are provided a codebase of value that is stable and secure.**
+
+## Policy
+
+* Versioning of this project will be idiomatic of a Go project using [Go
+  modules](https://github.com/golang/go/wiki/Modules).
+    * [Semantic import
+      versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning)
+      will be used.
+        * Versions will comply with [semver
+          2.0](https://semver.org/spec/v2.0.0.html) with the following exceptions.
+            * New methods may be added to exported API interfaces. All exported
+              interfaces that fall within this exception will include the following
+              paragraph in their public documentation.
+
+              > Warning: methods may be added to this interface in minor releases.
+
+        * If a module is version `v2` or higher, the major version of the module
+          must be included as a `/vN` at the end of the module paths used in
+          `go.mod` files (e.g., `module go.opentelemetry.io/collector/v2`, `require
+          go.opentelemetry.io/collector/v2 v2.0.1`) and in the package import path
+          (e.g., `import "go.opentelemetry.io/collector/v2/component"`). This includes the
+          paths used in `go get` commands (e.g., `go get
+          go.opentelemetry.io/collector/v2@v2.0.1`.  Note there is both a `/v2` and a
+          `@v2.0.1` in that example. One way to think about it is that the module
+          name now includes the `/v2`, so include `/v2` whenever you are using the
+          module name).
+        * If a module is version `v0` or `v1`, do not include the major version in
+          either the module path or the import path.
+    * Modules will be used to encapsulate signals and components.
+        * Experimental modules still under active development will be versioned at
+          `v0` to imply the stability guarantee defined by
```

Comment:
The discussions above https://github.com/open-telemetry/opentelemetry-collector/pull/3476/files#r656205029 suggests to stay on v0 even for APIs which we want to declare stable. How will we reconcile that statement with this line (they seems to be at odds)?

Reply:
To the extent that the core repository will remain a single* module, I think this entire section can be deleted.  I will replace it with a description of the use of modules to provide separation for internal tooling and note that all modules should be versioned together.